index.js

```diff
@@ -40,19 +46,36 @@ if (argv.dictionary) {
    * Humans do mistakes, so we allow several attempts
    * to enter the code.
    */
-  var attemptCounts = 0;
-  while (true) {
+  let attemptCounts = 0;
+  /**
+   * Token will expire when token timer reaches
+   * the max alloted time, in seconds.
+   */
+  let tokenTimer = 0;
+  /**
+   * Creates an object which calls checkFunction once
+   * a second. Is terminated with clearInterval(timerInterval);
+   */
+  const timerInterval = setInterval(checkToken, 1000);
+
+  function checkToken() {
     const attempt = prompt('Please enter token: ').trim();
 
     if (tokenGenerator.verify(attempt, tokenBundle.secret, tokenBundle.salt)) {
       console.log('Token has been successfully verified');
-      break;
+      clearInterval(timerInterval);
     }
 
-    attemptCounts++;
+    attemptCounts += 1;
     if (attemptCounts >= maxAttempts) {
       console.log('Verification failed! Too many attempts!');
-      break;
+      clearInterval(timerInterval);
+    }
+
+    tokenTimer += 1;
+    if (tokenTimer >= expirationTime) {
```

Comment:
Token validity should be checked before verifying it. With this code, if I enter correct token way after it expired, it would still pass.